modules/globebrowsing/src/globerotation.cpp

```diff
@@ -0,0 +1,236 @@
+        "This means that the object will be rotated to lay flat on the surface at the "
+        "given coordinate and follow the shape of the landscape."
+    };
+
+    struct [[codegen::Dictionary(GlobeRotation)]] Parameters {
+        // [[codegen::verbatim(GlobeInfo.description)]]
+        std::string globe
+            [[codegen::annotation("A valid scene graph node with a RenderableGlobe")]];
+
+        // [[codegen::verbatim(LatitudeInfo.description)]]
+        std::optional<double> latitude;
+
+        // [[codegen::verbatim(LongitudeInfo.description)]]
+        std::optional<double> longitude;
+
+        // [[codegen::verbatim(AngleInfo.description)]]
+        std::optional<double> angle;
+
+        // [[codegen::verbatim(UseHeightmapInfo.description)]]
+        std::optional<bool> useHeightmap;
+    };
+#include "globerotation_codegen.cpp"
+} // namespace
+
+namespace openspace::globebrowsing {
+
+documentation::Documentation GlobeRotation::Documentation() {
+    return codegen::doc<Parameters>("globebrowsing_rotation_globerotation");
+}
+
+GlobeRotation::GlobeRotation(const ghoul::Dictionary& dictionary)
+    : _globe(GlobeInfo)
+    , _latitude(LatitudeInfo, 0.0, -90.0, 90.0)
+    , _longitude(LongitudeInfo, 0.0, -180.0, 180.0)
+    , _angle(AngleInfo, 0.0, 0.0, 360.0)
+    , _useHeightmap(UseHeightmapInfo, false)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _globe = p.globe;
+    _globe.onChange([this]() {
+        findGlobe();
+        _matrixIsDirty = true;
+    });
+
+    _latitude = p.latitude.value_or(_latitude);
+    _latitude.onChange([this]() { _matrixIsDirty = true; });
+    addProperty(_latitude);
+
+    _longitude = p.longitude.value_or(_longitude);
+    _longitude.onChange([this]() { _matrixIsDirty = true; });
+    addProperty(_longitude);
+
+    _useHeightmap = p.useHeightmap.value_or(_useHeightmap);
+    _useHeightmap.onChange([this]() { _matrixIsDirty = true; });
+    addProperty(_useHeightmap);
+
+    _angle = p.angle.value_or(_angle);
+    _angle.onChange([this]() { _matrixIsDirty = true; });
+    addProperty(_angle);
+}
+
+void GlobeRotation::findGlobe() {
+    SceneGraphNode* n = sceneGraphNode(_globe);
+    if (n->renderable() && dynamic_cast<RenderableGlobe*>(n->renderable())) {
+        _globeNode = dynamic_cast<RenderableGlobe*>(n->renderable());
+    }
+    else {
+        LERRORC(
+            "GlobeRotation",
+            "Could not set attached node as it does not have a RenderableGlobe"
+        );
+        if (_globeNode) {
+            // Reset the globe name to it's previous name
+            _globe = _globeNode->identifier();
+        }
+    }
+}
+
+glm::vec3 GlobeRotation::computeSurfacePosition(double latitude, double longitude) const
+{
+    ghoul_assert(_globeNode, "Globe cannot be nullptr");
+
+    GlobeBrowsingModule& mod = *(global::moduleEngine->module<GlobeBrowsingModule>());
```

Comment:
Might be a bit more readable if:
`GlobeBrowsingModule* mod = global::module...`
and then using `->` instead of `.` below